### PR TITLE
Add tutorial + example SDF for shadow texture size

### PR DESCRIPTION
**Pull request by athenaz2**
# 🎉 New feature

Add tutorial + example SDF. Related to https://github.com/gazebosim/gz-rendering/pull/1034

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: examples/worlds/shadow_texture_size.sdf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      <ambient>0.0 0.0 0.0</ambient>
+    </scene>
+
+    <!--GUI plugins-->
+    <gui>
+      <plugin filename="MinimalScene" name="3D View">
+        <engine>ogre2</engine>
+        <camera_pose>-10 0 7 0 0.5 0</camera_pose>
+        <shadows>
+          <texture_size light_type="directional">16384</texture_size>
```
Comment (iche033) [inline examples/worlds/shadow_texture_size.sdf:15]: set this to `8192` as I see that's what the tutorial tells the user to do?

File: tutorials/shadow_texture_size.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+## Changing shadow texture size
+
+The shadow texture size can be changed for the GUI with the usage of the Minimal Scene GUI plugin in the SDF.
+
+### Example usage for the GUI
+
+Let's change the shadow texture size for directional light with the SDF file below. (The finished SDF file can be viewed [here](https://github.com/gazebosim/gz-sim/blob/main/examples/worlds/shadow_texture_size.sdf).)
+
+1) Save the below in an SDF file named `shadow_texsize_demo.sdf`:
```
Comment (iche033) [inline tutorials/shadow_texture_size.md:23]: Change the name to be consistent with finished SDF file in `examples/worlds`?

```suggestion
1) Save the below in an SDF file named `shadow_texture_size.sdf`:
```

File: tutorials/shadow_texture_size.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  <texture_size light_type="directional">8192</texture_size>
+</shadows>
+```
+
+The `<texture_size>` value can be changed as per the Texture size options section.
+
+3) Open the `shadow_texsize_demo.sdf` world with
+
+```bash
+gz sim shadow_texsize_demo.sdf
```
Comment (iche033) [inline tutorials/shadow_texture_size.md:111]:
```suggestion
gz sim shadow_texture_size.sdf
```

File: tutorials/shadow_texture_size.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+The default shadow texture size for all lights is 2K. The default max texture size is 16K, but for less powerful computers, it may be a lower value, like 8K.
+
+## Changing shadow texture size
+
+The shadow texture size can be changed for the GUI with the usage of the Minimal Scene GUI plugin in the SDF.
+
+### Example usage for the GUI
+
+Let's change the shadow texture size for directional light with the SDF file below. (The finished SDF file can be viewed [here](https://github.com/gazebosim/gz-sim/blob/main/examples/worlds/shadow_texture_size.sdf).)
```
Comment (iche033) [inline tutorials/shadow_texture_size.md:21]: can you mention that we currently only support changing shadow texture size for directional lights?

File: tutorials/shadow_texture_size.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+This tutorial showcases how to change the texture size of shadows casted by lights.
+
+<div style="text-align:center;">
+  \image html files/shadow_texture_size/shadow_texsize_2k_to_16k.gif width=60%
+</div>
+
+## Texture size options
+
+Supported texture sizes include: 512 px, 1024 px, 2048 px, 4096 px, 8192 px, 16384 px.
```
Comment (iche033) [inline tutorials/shadow_texture_size.md:11]:
```suggestion
Supported texture sizes include: 512px, 1024px, 2048px, 4096px, 8192px, 16384px.
```

File: tutorials/shadow_texture_size.md (hunk 4 above)
Comment (athenaz2) [reply tutorials/shadow_texture_size.md:23]: Yup = done here b48725b50d1b4fd554ec15e56d9c8e9d8d6c6823

File: examples/worlds/shadow_texture_size.sdf (hunk 1 above)
Comment (athenaz2) [reply examples/worlds/shadow_texture_size.sdf]: 0d21682c904daa3af7f3c5f860e9eee75f3075f8

File: tutorials/shadow_texture_size.md (hunk 4 above)
Comment (iche033) [reply tutorials/shadow_texture_size.md:23]: oh can you also change the link to point to gz-sim9 branch? i.e. https://github.com/gazebosim/gz-sim/tree/gz-sim9/examples/worlds/shadow_texture_size.sdf

**Review by iche033 — APPROVED** [review]

**Pull request merged**